src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -819,6 +825,8 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
                         source.getPathForExceptionMessage(),
                         target.getPathForExceptionMessage(),
                         x.errorString());
+                } else if (x.errno() == ENOTDIR) {
+                    x.setError(ENOENT);
```

Comment:
This is rename where the issue might be the source or target path. So I think we have to drop this change too.

Reply:
Reverted in 0908e75.